src/modules/exec.ts

```diff
@@ -135,3 +138,31 @@ export async function runExecSpec(
         throw error;
     }
 }
+
+async function ensureVenvIfPython(
+    command: string,
+    envVars: NodeJS.ProcessEnv,
+    cwd: string,
+    loggingOptions: { writeStdout?: boolean; verbose?: boolean },
+) {
+    if (['pip', 'pip3', 'python', 'python3'].includes(command)) {
+        let venvDir = `${envVars.VELOCITAS_CACHE_DIR}/pyvenv/${envVars.builtin_component_id}`;
```

Comment:
All of the variables you are extracing from the envVars should be readily available in better and typed forms rather than pulling it out of the untyped envVars.
e.g.
`ProjectCache.getCacheDir()` and `componentContext.manifest.id`